gossip/c_block_callbacks.go

```diff
@@ -233,11 +237,25 @@ func consensusCallbackBeginBlockFn(
 					}
 
 					block, blockEvents := spillBlockEvents(store, block, es.Rules)
-					txs := make(types.Transactions, 0, blockEvents.Len()*10)
+					unorderedTxs := make([]ScramblerEntry, 0, blockEvents.Len()*10)
+
+					signer := gsignercache.Wrap(types.MakeSigner(chainCfg, blockNumber, uint64(blockCtx.Time)))
 					for _, e := range blockEvents {
-						txs = append(txs, e.Txs()...)
+						for _, tx := range e.Txs() {
+							entry, err := newScramblerTransaction(signer, tx)
+							if err != nil {
+								log.Crit(fmt.Sprintf("cannot create scrambler tx %s", tx.Hash()), "err", err)
```

Comment:
Please handle this without crashing the system. `log.Crit` will terminate the process.

Reply:
Im not sure how? This func does not return an error. If an error occurs in this part of the app, its always a fatal one...

Reply:
Discussed in private. Understood now